docker/Dockerfile.dev

```diff
@@ -0,0 +1,25 @@
+FROM python:3.7.2-alpine3.8
+
+ENV PYTHONUNBUFFERED=1
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PATH "/root/.poetry/bin:$PATH"
+
+WORKDIR /srv/app
+
+COPY ./pyproject.toml ./
+COPY ./poetry.lock ./
+
+RUN apk add --no-cache \
```

Comment:
Previous comment, which vanished when the diff changes:
```


I think this RUN should be moved before the previous COPY * because it can be cached even if pyproject.toml or poetry.lock change; in this moment it will be re-executed for any change to these two files, but it doesn't need to. If a Python package is added that requires a new OS package, the line itself will change and purge its own cache.
```
so I think the sequence should be:
1. `RUN` with all `apk` commands (includind `apk del`) and `curl | python` for `poetry`: they change very slowly, so don't need to be reinstalled whenever a Python package dependency changes; possibly the `poetry config` command can be attached as it's not dependent on any project file too
2. single `COPY` for `pyproject.toml` and `poetry.lock`: `COPY` supports multiple arguments. This starts the section that depends on project files and whose cache will be invalidated when these two files change
3. `RUN` for `poetry install`

You may want to split 1 into `apk`-related and `poetry`-related if you don't want to reinstall `poetry` when the list of OS packages change. I don't think the two interact much so that should not increase the image/layer size more than a single `RUN`.

Reply:
Rewriting this:
- `bash`, `grep` can go up because they are not needed by `poetry install`
- `curl -sSL ... | python` can also go up before project-specific files

Reply:
updated.